commit 6804b3ecbe20654891324ef82f840ba02d642d6c
Author: David Rientjes <rientjes@google.com>
Date:   Wed Jan 2 12:56:33 2019 -0800

    kvm: sev: Fail KVM_SEV_INIT if already initialized
    
    [ Upstream commit 3f14a89d1132dcae3c8ce6721c6ef51f6e6d9b5f ]
    
    By code inspection, it was found that multiple calls to KVM_SEV_INIT
    could deplete asid bits and overwrite kvm_sev_info's regions_list.
    
    Multiple calls to KVM_SVM_INIT is not likely to occur with QEMU, but this
    should likely be fixed anyway.
    
    This code is serialized by kvm->lock.
    
    Fixes: 1654efcbc431 ("KVM: SVM: Add KVM_SEV_INIT command")
    Reported-by: Cfir Cohen <cfir@google.com>
    Signed-off-by: David Rientjes <rientjes@google.com>
    Signed-off-by: Radim Krčmář <rkrcmar@redhat.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/arch/x86/kvm/svm.c b/arch/x86/kvm/svm.c
index 02ac8fa0cd6d..ee8f8d70b98a 100644
--- a/arch/x86/kvm/svm.c
+++ b/arch/x86/kvm/svm.c
@@ -6256,6 +6256,9 @@ static int sev_guest_init(struct kvm *kvm, struct kvm_sev_cmd *argp)
 	int asid, ret;
 
 	ret = -EBUSY;
+	if (unlikely(sev->active))
+		return ret;
+
 	asid = sev_asid_new();
 	if (asid < 0)
 		return ret;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1402_6804b3ecbe20654891324ef82f840ba02d642d6c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1402_6804b3ecbe20654891324ef82f840ba02d642d6c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
